syntax = "proto2";

message AuthRequest {
    required int64 user_id = 1;
}

message AuthUrl {
    required int64 user_id = 1;
    required string auth_url = 2;
}

message AuthCode {
    required int64 user_id = 1;
    required string auth_code = 2;
}

message AuthStatus {
    required int64 user_id = 1;
    required bool ok = 2;
}

message CalendarEvent {
    required int64 user_id = 1;
    required int64 meeting_id = 2;
    required string desc = 3;
    required int64 time = 4;
}

message EventAdded {
    required bool added = 1;
}

service CalendarService {
    rpc RequestAuth (AuthRequest) returns (AuthUrl);
    rpc FinishAuth (AuthCode) returns (AuthStatus);
    rpc PushEvent (CalendarEvent) returns (EventAdded);
}